File: javascript/easy/isomorphicStrings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    let map = {}, reverseMap = {};
    for(let i = 0; i < s.length; i++) {
        let c1 = s[i], c2= t[i];
        if((map[c1] !== undefined && map[c1] !== c2) || (reverseMap[c2] !== undefined && reverseMap[c2] !== c1)) {
           return false;
        }
        map[c1] = c2;
        reverseMap[c2] = c1;
    }
    return true;
};

File: ruby/easy/isomorphic_strings.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_isomorphic(s, t)
    map, reverse_map = {}, {}
    for i in 0..s.length-1
        c1, c2= s[i].to_sym, t[i].to_sym
        return false if((!map[c1].nil? && map[c1] != c2) || (!reverse_map[c2].nil? && reverse_map[c2] != c1))
        map[c1] = c2
        reverse_map[c2] = c1
    end
    true
end